webdriver/tests/bidi/browsing_context/close/prompt_unload.py

```diff
@@ -0,0 +1,112 @@
+import pytest
+import asyncio
+
+from webdriver.bidi.modules.script import ContextTarget
+
+pytestmark = pytest.mark.asyncio
+
+CONTEXT_DESTROYED_EVENT = "browsingContext.contextDestroyed"
+USER_PROMPT_OPENED_EVENT = "browsingContext.userPromptOpened"
+
+
+@pytest.mark.parametrize("type_hint", ["window", "tab"])
+@pytest.mark.parametrize("prompt_unload", [None, False])
+async def test_prompt_unload_not_triggering_dialog(bidi_session, subscribe_events, url, wait_for_event,
+                                                   wait_for_future_safe, type_hint, prompt_unload):
+
+    new_context = await bidi_session.browsing_context.create(type_hint=type_hint)
+
+    page_beforeunload = url(
+        "/webdriver/tests/support/html/beforeunload.html")
```

Comment:
We now have a `setup_beforeunload_page` fixture that you should use here. It does all the necessary steps including the user activity to mark the page as dirty. As such you can remove the actions below.

Please apply that to the other tests in this file as well.